make distcheck fails

```
$ make distcheck
...
make[3]: Entering directory '/home/edg/src/arkimet/arkimet-1.0/_build/sub/arki'
../../../arki/buildflex ../../../arki/matcher/reftime/reftime-lex.ll
flex: could not create reftime-lex.cc
Makefile:4068: recipe for target '../../../arki/matcher/reftime/reftime-lex.h' failed
make[3]: *** [../../../arki/matcher/reftime/reftime-lex.h] Error 1
make[3]: Leaving directory '/home/edg/src/arkimet/arkimet-1.0/_build/sub/arki'
Makefile:614: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory '/home/edg/src/arkimet/arkimet-1.0/_build/sub'
Makefile:503: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/edg/src/arkimet/arkimet-1.0/_build/sub'
Makefile:820: recipe for target 'distcheck' failed
make: *** [distcheck] Error 1
```

--- a/Makefile.am
+++ b/Makefile.am
@@ -37,6 +37,7 @@ EXTRA_DIST = \
 	     README.Cineca \
 	     NEWS.md \
 	     update-tags \
-	     autogen.sh
+	     autogen.sh \
+	     run-local
 
 CLEANFILES = $(man_MANS)

--- a/Makefile.am
+++ b/Makefile.am
@@ -23,11 +23,11 @@ man_MANS += arki-bufr-prepare.1
 endif
 
 arki-server.1: src/arki-server
-	COLUMNS=200 ./run-local help2man --name='Arkimet server for remote dataset access' --section=1 --no-info --version-string="$(PACKAGE_VERSION)" $< > $@ || rm -f $@
+	COLUMNS=200 $(srcdir)/run-local help2man --name='Arkimet server for remote dataset access' --section=1 --no-info --version-string="$(PACKAGE_VERSION)" $< > $@ || rm -f $@
 
 
 arki-%.1: src/arki-% doc/mhooks-%
-	./run-local $< --manpage="$(word 2,$^)" > $@ || rm -f $@
+	$(srcdir)/run-local $< --manpage="$(word 2,$^)" > $@ || rm -f $@
 
 #%.1: tools/%
 #	( cd tools && ./`basename $<` help manpage ) > $@

--- a/arki/Makefile.am
+++ b/arki/Makefile.am
@@ -188,11 +188,14 @@ BUILT_SOURCES = $(reftime_parser_built_sources)
 
 lib_LTLIBRARIES = libarkimet.la
 
+noinst_LTLIBRARIES = libarkimet-test.la
+
 # Build these incrementally
 libarkimet_la_CPPFLAGS = $(AM_CPPFLAGS) -Werror
 libarkimet_la_SOURCES =
 libarkimet_la_LDFLAGS = -version-info @LIBARKI_VERSION_INFO@ $(SQLITE3_LDFLAGS) $(ZLIB_LIBS) $(DEVEL_LIBS) -lpthread
-tests_arki_test_SOURCES = tests/tut-main.cpp
+libarkimet_test_la_SOURCES = tests/tut-main.cpp
+tests_arki_test_SOURCES =
 tests_arki_test_LDADD =
 
 
@@ -281,13 +284,15 @@ libarkimet_la_SOURCES += \
 		runtime/io.cc \
 		types-init.cc
 
-tests_arki_test_SOURCES += \
+libarkimet_test_la_SOURCES += \
 		tests/tests.cc \
-		tests/tests-tut.cc \
-		core/time-test.cc \
 		core/tests.cc \
 		types/tests.cc \
-		utils/tests.cc \
+		utils/tests.cc
+
+tests_arki_test_SOURCES += \
+		tests/tests-tut.cc \
+		core/time-test.cc \
 		utils-test.cc \
 		utils/accounting-tut.cc \
 		utils/gzip-test.cc \
@@ -608,7 +613,7 @@ check-local:
 		$(TESTS_ENVIRONMENT) $$test ; \
 	done
 
-tests_arki_test_LDADD += libarkimet.la $(SQLITE3_LDFLAGS) $(DEVEL_LIBS) -lpthread
+tests_arki_test_LDADD += libarkimet.la libarkimet-test.la $(SQLITE3_LDFLAGS) $(DEVEL_LIBS) -lpthread
 
 EXTRA_DIST = runtest buildbison buildflex \
 	     matcher/reftime/reftime-lex.ll \

--- a/arki/Makefile.am
+++ b/arki/Makefile.am
@@ -94,6 +94,7 @@ nobase_dist_arkiinclude_HEADERS = \
 		matcher/timerange.h \
 		matcher/utils.h \
 		metadata/collection.h \
+		metadata/consumer.h \
 		metadata/clusterer.h \
 		metadata/xargs.h \
 		metadatagrid.h \
@@ -371,6 +372,7 @@ libarkimet_la_SOURCES += \
 		metadata/collection.cc \
 		metadata/stream.cc \
 		metadata/clusterer.cc \
+		metadata/consumer.cc \
 		metadata/test-generator.cc \
 		summary/stats.cc \
 		summary/intern.cc \

--- a/arki/Makefile.am
+++ b/arki/Makefile.am
@@ -169,6 +169,7 @@ nobase_dist_arkiinclude_HEADERS = \
 		utils/commandline/options.h \
 		utils/commandline/parser.h \
 		utils/commandline/wordwrap.h \
+		utils/tests.h \
 		values.h
 
 reftime_parser_built_sources = \

--- a/arki/Makefile.am
+++ b/arki/Makefile.am
@@ -170,13 +170,13 @@ nobase_dist_arkiinclude_HEADERS = \
 		utils/commandline/wordwrap.h \
 		values.h
 
-BUILT_SOURCES = \
-		$(srcdir)/matcher/reftime/reftime-lex.h \
-		$(srcdir)/matcher/reftime/reftime-lex.cc \
-		$(srcdir)/matcher/reftime/reftime-parse.hh \
-		$(srcdir)/matcher/reftime/reftime-parse.cc
+reftime_parser_built_sources = \
+		matcher/reftime/reftime-lex.h \
+		matcher/reftime/reftime-lex.cc \
+		matcher/reftime/reftime-parse.hh \
+		matcher/reftime/reftime-parse.cc
 
-all_built_sources: $(BUILT_SOURCES)
+BUILT_SOURCES = $(reftime_parser_built_sources)
 
 %-lex.h %-lex.cc: %-lex.ll
 	$(srcdir)/buildflex $<
@@ -610,4 +610,4 @@ tests_arki_test_LDADD += libarkimet.la $(SQLITE3_LDFLAGS) $(DEVEL_LIBS) -lpthrea
 EXTRA_DIST = runtest buildbison buildflex \
 	     matcher/reftime/reftime-lex.ll \
 	     matcher/reftime/reftime-parse.yy \
-	     $(all_built_sources)
+	     $(reftime_parser_built_sources)

--- a/arki/runtest
+++ b/arki/runtest
@@ -74,7 +74,7 @@ cp -a "$TOP_SRCDIR/test/misc" ./
 cp -a "$TOP_SRCDIR/doc/lua-examples" ./
 cp -a "$TOP_SRCDIR/doc/matcher" ./
 # Ensure that the directory can be removed at cleanup
-chmod u+w -R .
+chmod u+w -R "$TESTDIR"
 # Create test dataset directories
 mkdir test200
 mkdir test80

--- a/arki/runtest
+++ b/arki/runtest
@@ -73,6 +73,8 @@ cp -a "$TOP_SRCDIR/test/postproc" ./
 cp -a "$TOP_SRCDIR/test/misc" ./
 cp -a "$TOP_SRCDIR/doc/lua-examples" ./
 cp -a "$TOP_SRCDIR/doc/matcher" ./
+# Ensure that the directory can be removed at cleanup
+chmod u+w -R .
 # Create test dataset directories
 mkdir test200
 mkdir test80

--- a/python/Makefile.am
+++ b/python/Makefile.am
@@ -14,7 +14,7 @@ _arkimet_la_SOURCES = \
     summary.cc \
     dataset.cc \
     arkimet.cc
-_arkimet_la_CPPFLAGS = $(PYTHON_CFLAGS) -I$(top_srcdir)
+_arkimet_la_CPPFLAGS = $(PYTHON_CFLAGS) -I$(top_srcdir) -I$(top_builddir)
 _arkimet_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_arkimet
 _arkimet_la_LIBADD = ../arki/libarkimet.la
 

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -66,11 +66,7 @@ check-local:
 		$(srcdir)/$$test ; \
 	done
 
-test_src_SOURCES = \
-		arki-server-test.cc \
-		../arki/utils/tests.cc \
-		../arki/types/tests.cc \
-		../arki/tests/tut-main.cpp
-test_src_LDADD = $(all_libs)
+test_src_SOURCES = arki-server-test.cc
+test_src_LDADD = $(all_libs) ../arki/libarkimet-test.la
 
 EXTRA_DIST = testlib.sh $(test_scripts)

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -62,8 +62,8 @@ test_scripts = test-arki-scan test-arki-query test-arki-xargs test-arki-check ru
 check_PROGRAMS = test-src
 
 check-local:
-	for test in $(srcdir)/$(test_scripts); do \
-		./$$test ; \
+	for test in $(test_scripts); do \
+		$(srcdir)/$$test ; \
 	done
 
 test_src_SOURCES = \

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
 bin_PROGRAMS = arki-check arki-query arki-dump arki-gzip arki-mergeconf arki-scan arki-xargs
-bin_SCRIPTS = arki-server
+dist_bin_SCRIPTS = arki-server
 noinst_PROGRAMS = arki-test
 
 dist_noinst_HEADERS = bench/benchmark.h

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -62,7 +62,7 @@ test_scripts = test-arki-scan test-arki-query test-arki-xargs test-arki-check ru
 check_PROGRAMS = test-src
 
 check-local:
-	for test in $(test_scripts); do \
+	for test in $(srcdir)/$(test_scripts); do \
 		./$$test ; \
 	done
 

--- a/src/test-arki-scan
+++ b/src/test-arki-scan
@@ -15,7 +15,7 @@ echo -n "Testing arki-scan"
 runtest "arki-scan --yaml inbound/test.grib1 | grep -q ^Origin"
 
 # Try scanning a BUFR file
-if grep -q "HAVE_DBALLE 1" $TOP_SRCDIR/config.h
+if grep -q "HAVE_DBALLE 1" $TOP_BUILDDIR/config.h
 then
 	runtest "arki-scan --yaml inbound/test.bufr | grep -q ^Origin"
 fi

--- a/src/testlib.sh
+++ b/src/testlib.sh
@@ -21,6 +21,7 @@ setup() {
     fi
 
     TOP_SRCDIR=$(readlink -f $(pwd)/$(dirname $0)/..)
+    TOP_BUILDDIR=$(readlink -f ..)
     BINDIR=$TOP_SRCDIR/src
 
     export PATH="$BINDIR:$PATH"

--- a/src/testlib.sh
+++ b/src/testlib.sh
@@ -58,6 +58,7 @@ setup() {
     mkdir inbound
     # Put data in test inbound area
     cp -a "$TOP_SRCDIR/test/data/"* inbound/
+    chmod u+w -R "$TESTDIR"
     # Create test dataset directories
     mkdir test200
     mkdir test80

--- a/src/testlib.sh
+++ b/src/testlib.sh
@@ -24,7 +24,7 @@ setup() {
     TOP_BUILDDIR=$(readlink -f ..)
     BINDIR=$TOP_BUILDDIR/src
 
-    export PATH="$TOP_SRCDIR/src:$TOP_BINDIR/src:$PATH"
+    export PATH="$TOP_SRCDIR/src:$TOP_BUILDDIR/src:$PATH"
 
     CMD=`pwd`/"$1"
 

--- a/src/testlib.sh
+++ b/src/testlib.sh
@@ -22,9 +22,9 @@ setup() {
 
     TOP_SRCDIR=$(readlink -f $(pwd)/$(dirname $0)/..)
     TOP_BUILDDIR=$(readlink -f ..)
-    BINDIR=$TOP_SRCDIR/src
+    BINDIR=$TOP_BUILDDIR/src
 
-    export PATH="$BINDIR:$PATH"
+    export PATH="$TOP_SRCDIR/src:$TOP_BINDIR/src:$PATH"
 
     CMD=`pwd`/"$1"
 

--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -15,8 +15,12 @@ EXTRA_DIST = \
 	     dataset1/config \
 	     misc/erse00.expa \
 	     misc/rec-ts-alias.conf \
+	     postproc/cat \
 	     postproc/checkfiles \
+	     postproc/countbytes \
 	     postproc/error \
 	     postproc/null \
+	     postproc/outthenerr \
 	     postproc/say \
-	     postproc/testcountbytes
+	     postproc/testcountbytes \
+	     postproc/zeroes

